lib/maths/CBoostedTreeImpl.cc

```diff
@@ -725,11 +737,18 @@ bool CBoostedTreeImpl::selectNextHyperparameters(const TMeanVarAccumulator& loss
     if (m_FeatureBagFractionOverride == boost::none) {
         parameters(i++) = m_FeatureBagFraction;
     }
+    LOG_TRACE(<< "loss moments = " << lossMoments << " for lambda = " << m_Lambda
+              << ", gamma = " << m_Gamma << ", eta = " << m_Eta
+              << ", eta growth rate per tree = " << m_EtaGrowthRatePerTree
+              << ", feature bag fraction = " << m_FeatureBagFraction);
 
     double meanLoss{CBasicStatistics::mean(lossMoments)};
-    double lossVariance{CBasicStatistics::variance(lossMoments)};
+    double meanLossVariance{CBasicStatistics::variance(lossMoments) /
+                            CBasicStatistics::count(lossMoments)};
```

Comment:
It is unclear, why you need to divide `variance` by `count`. Maybe you can write a comment explaining your decision.

Reply:
I addressed this in [c2220ef](https://github.com/elastic/ml-cpp/pull/625/commits/c2220ef0f01b826a22824b4dcd9774ddc6cd18b7). Does this seem reasonable to you?